buildtest/buildsystem/schemas/utils.py

```
@@ -0,0 +1,112 @@
+"""
+Utility and helper functions for schemas.
+Copyright (C) 2020 Vanessa Sochat.
+"""
+
+import os
+import json
+from jsonschema import validate
+import re
+import sys
+import yaml
+
+here = os.path.dirname(os.path.abspath(__file__))
+
+
+def load_schema(path):
+    """load a schema from file. We assume a json file
+
+       Parameters
+       ==========
+       path: the path to the schema file
+    """
+    if not os.path.exists(path):
+        sys.exit("schema file %s does not exist." % path)
+    with open(path, "r") as fd:
+        schema = json.loads(fd.read())
+    return schema
+
+
+def load_recipe(path):
+    """load a yaml recipe file. The recipe is validated against a schema.
+    """
+    if not os.path.exists(path):
+        sys.exit("test configuration file %s does not exist." % path)
+    with open(path, "r") as fd:
+        content = yaml.load(fd.read(), Loader=yaml.SafeLoader)
+    return content
+
+
+def get_latest(schema1, schema2):
+    """between two schema files, compare versions and determine which is latest
+    """
+    schema1_version = get_schema_version(schema1)
+    schema2_version = get_schema_version(schema2)
+
+    # Convert version to int to determine latest
+    if int(schema1_version.replace(".", "")) > int(schema2_version.replace(".", "")):
+        return schema1
+    return schema2
+
+
+def get_schema_version(schema):
+    """Given a schema file, use the convention naming of
+       <name>-v<version>.schema.json to derive the version
+    """
+    match = re.search(
+        "v(?P<version>[0-9]{1}[.][0-9]{1}[.][0-9]{1})[.]schema[.]json", schema
+    )
+    if match:
```

just a thought if we don't have a match ``if match`` what do we want to return considering ``get_schema_version`` is used by ``get_latest`` to calculate latest version. If there is no return value then ``get_latest`` won't calculate. I assume it is unlikely since we are populating the schema files. Maybe raise an error would be best.

---

A schema that is validated / tested from the schemas repository must conform to that file name pattern.